### PR TITLE
Prepare docs for sub-packages

### LINKED ISSUE #94: Figure out how to make docs of `cuda.core` and `cuda.bindings` coexist
Main question: Do we want to the same project docs with `cuda.core` (#70) & `cuda.bindings` (#75) coexist say under separate tabs or pages, or do we want each namespace package to have their own doc pages (similar to `cuda.{parallel,cooperative}`)?

### LINKED ISSUE #95: Enable versioned doc support

### PR DESCRIPTION
**Pull request by leofang**
Close #94. Close #95.

This PR prepares for the documentation infrastructure for adding sub-packages. This introduces a (hopefully one-time only) breaking change to the existing doc urls. Two major changes are:
- cuda-python becomes a landing page for sub-packages (to prepare for #105)
- subdirectories are used for sub-packages' docs
- multi-version doc support is added

Old urls:
- https://nvidia.github.io/cuda-python/
   - https://nvidia.github.io/cuda-python/overview.html
   - https://nvidia.github.io/cuda-python/install.html
   - https://nvidia.github.io/cuda-python/release.html
   - ...

New urls:
- https://nvidia.github.io/cuda-python/
   - will redirects to https://nvidia.github.io/cuda-python/latest/
      - which will be a copy of https://nvidia.github.io/cuda-python/12.6.1/ as of today
   - https://nvidia.github.io/cuda-python/latest/release.html
   - ...
   - https://nvidia.github.io/cuda-python/12.6.1/release.html
   - ...
- https://nvidia.github.io/cuda-python/cuda-bindings
   - will redirects to https://nvidia.github.io/cuda-python/cuda-bindings/latest/
      - which will be a copy of https://nvidia.github.io/cuda-python/cuda-bindings/12.6.1/ as of today
   - https://nvidia.github.io/cuda-python/cuda-bindings/latest/overview.html
   - https://nvidia.github.io/cuda-python/cuda-bindings/latest/install.html
   - https://nvidia.github.io/cuda-python/cuda-bindings/latest/release.html
   - ...
   - https://nvidia.github.io/cuda-python/cuda-bindings/12.6.1/overview.html
   - https://nvidia.github.io/cuda-python/cuda-bindings/12.6.1/install.html
   - https://nvidia.github.io/cuda-python/cuda-bindings/12.6.1/release.html
   - ...
- https://nvidia.github.io/cuda-python/cuda-core
   - will redirects to https://nvidia.github.io/cuda-python/cuda-core/latest/
      - which will be a copy of https://nvidia.github.io/cuda-python/cuda-core/0.1.0/ as of today
   - https://nvidia.github.io/cuda-python/cuda-core/latest/install.html
   - https://nvidia.github.io/cuda-python/cuda-core/latest/release.html
   - ...
   - https://nvidia.github.io/cuda-python/cuda-core/0.1.0/install.html
   - https://nvidia.github.io/cuda-python/cuda-core/0.1.0/release.html
   - ...

### CODE REVIEW

**Comment by leofang** [comment]
This should be ready for review. However, do not merge yet, as I need to do a demo first before we commit to the changes.

**Comment by leofang** [comment]
(Updated the PR description to compare the old/existing urls with the new/expected urls.)

File: cuda_bindings/docs/environment-docs.yml
Comment (vzhurba01) [inline cuda_bindings/docs/environment-docs.yml:1]: Can we move this file to cuda_python instead? I'd like to have a single yml that's configured to support doc builds across all subpackages.

Comment (leofang) [reply cuda_bindings/docs/environment-docs.yml:1]: Done in 75305fb. I wasn't sure if it should be at root directory or in `cuda_python/docs`. I think either way is fine.

File: cuda_python/docs/build_all_docs.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+./build_docs.sh
+
+# build cuda-bindings docs
+CUDA_BINDINGS_PATH=build/html/cuda-bindings
+mkdir -p $CUDA_BINDINGS_PATH
+pushd .
+cd ../../cuda_bindings/docs
+rm -rf build
+./build_docs.sh
+cp -r build/html/* "$(dirs +1)"/$CUDA_BINDINGS_PATH
```
Comment (vzhurba01) [inline cuda_python/docs/build_all_docs.sh]: >cp: target '~/tip/public/leofang/cuda-python/cuda_python/docs/build/html/cuda-bindings' is not a directory

Tilde expansion doesn't work for me. Can you replace it with `$(dirs -l +1)`?

File: cuda_python/docs/README.md
Comment (vzhurba01) [inline cuda_python/docs/README.md:1]: Can you expand this readme with how to use CUDA_PYTHON_DOMAIN?

In the earlier demo you had called `python -m http.server -d build/html/ 8787` and I'd like this step referenced in this readme.

File: cuda_python/docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+#   - https://stackoverflow.com/a/61694897/2344149
+#   - https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_epilog
+CUDA_PYTHON_DOMAIN = os.environ.get('CUDA_PYTHON_DOMAIN',
+                                    'https://nvidia.github.io/cuda-python')
+rst_epilog = f"""
+.. _cuda.core: {CUDA_PYTHON_DOMAIN}/cuda-core/latest
+.. _cuda.bindings: {CUDA_PYTHON_DOMAIN}/cuda-bindings/latest
+.. _cuda.cooperative: https://nvidia.github.io/cccl/cuda_cooperative/
+.. _cuda.parallel: https://nvidia.github.io/cccl/cuda_parallel/
+"""
```
Comment (vzhurba01) [inline cuda_python/docs/source/conf.py:90]: Testing this locally, it looks like this replacement doesn't get applied to all links. Is this just a sphinx limitation?

File: cuda_python/docs/build_all_docs.sh (hunk 1 above)
Comment (leofang) [reply cuda_python/docs/build_all_docs.sh]: Good catch, fixed in commit fa8d482. I was developing on a shared machine where I could not put large files under `~/`, thus missing this.

File: cuda_python/docs/README.md
Comment (leofang) [reply cuda_python/docs/README.md:1]: Addressed in commit https://github.com/NVIDIA/cuda-python/commit/fa8d48236f99b369325a99a47429475c0c34205a.

File: cuda_python/docs/source/conf.py (hunk 2 above)
Comment (leofang) [reply cuda_python/docs/source/conf.py:90]: Right. Anything on the left panel (belonging to the table of contents) are hard-wired in `index.rst`. Sphinx toctree is terribly bad, ex: https://github.com/sphinx-doc/sphinx/issues/701 (which would've helped us if fixed).

**Review by vzhurba01 — APPROVED** [review]
LGTM

File: cuda_core/docs/source/_templates/sidebar/variant-selector.html
Comment (leofang) [inline cuda_core/docs/source/_templates/sidebar/variant-selector.html:1]: Note: There's a reason that this file has to be placed in this funny path. Per the Furo docs this is the way to overwrite its default template:
- https://pradyunsg.me/furo/customisation/sidebar/#using-templates-path
- This is the file that we overwrite: https://github.com/pradyunsg/furo/blob/main/src/furo/theme/furo/sidebar/variant-selector.html

**Pull request merged**